verilog/transform/obfuscate.cc

```diff
@@ -141,6 +144,38 @@ static absl::Status VerifyEquivalence(absl::string_view original,
   return absl::OkStatus();
 }
 
+// Find all modules and collect interface names
+absl::Status CollectInterfaceNames(absl::string_view content,
+                                   verible::StringSet* if_names) {
+  VLOG(1) << __FUNCTION__;
+
+  const auto analyzer =
+      verilog::VerilogAnalyzer::AnalyzeAutomaticMode(content, NULL);
+  const auto lex_status = ABSL_DIE_IF_NULL(analyzer)->LexStatus();
+  const auto parse_status = analyzer->ParseStatus();
+
+  if (lex_status.ok() && parse_status.ok()) {
+    const auto& syntax_tree = analyzer->SyntaxTree();
+    const auto mod_headers =
+      FindAllModuleHeaders(*ABSL_DIE_IF_NULL(syntax_tree).get());
+
+    // For each module, collect all indentifiers under the module
+    // header tree into the StringSet.
+    for(auto const& mod_header: mod_headers) {
+      const auto if_leafs =
+        FindAllSymbolIdentifierLeafs(*mod_header.match);
```

Comment:
If `` `MACRO`` are referenced in headers, should those be preserved too (`MacroIdentifier`, `MacroCallId`, `MacroIdItem`)?

Reply:
In our flow, we expand macros before doing obfuscation so this does not affect our usage. However, I can imagine someone would define macros as their parameter (e.g. bit-width) and include an all-in-one `define.h` in the design. In this case, we may need another option (like `--preserve_macro`) to preserve macro related tokens as `define.h` may not be transformed together.

Should I make the modification with this PR, or we should open another issue for this?

Reply:
I'd say separate issue, separate PR.